cli/command/container/diff_test.go

```diff
@@ -0,0 +1,95 @@
+package container
+
+import (
+	"context"
+	"io"
+	"strings"
+	"testing"
+
+	"github.com/docker/cli/internal/test"
+	"github.com/docker/docker/api/types/container"
+	"github.com/pkg/errors"
+	"gotest.tools/v3/assert"
+)
+
+func TestRunDiff(t *testing.T) {
+	cli := test.NewFakeCli(&fakeClient{
+		containerDiffFunc: func(
+			ctx context.Context,
+			containerID string,
+		) ([]container.FilesystemChange, error) {
+			return []container.FilesystemChange{
+				{
+					Kind: 0, // Modified ("C")
+					Path: "/path/to/file0",
+				},
+				{
+					Kind: 1, //  Added ("A")
+					Path: "/path/to/file1",
+				},
+				{
+					Kind: 2, // Deleted ("D")
+					Path: "/path/to/file2",
+				},
+			}, nil
+		},
+	})
+
+	cmd := NewDiffCommand(cli)
+	cmd.SetOut(io.Discard)
+
+	containerId := "a61d6b418006"
+	cmd.SetArgs([]string{containerId})
+
+	err := cmd.Execute()
+	assert.Assert(t, err == nil)
+
+	diff := strings.SplitN(cli.OutBuffer().String(), "\n", 3)
+	file0 := strings.TrimSpace(diff[0])
+	file1 := strings.TrimSpace(diff[1])
+	file2 := strings.TrimSpace(diff[2])
+
+	assert.Assert(t, len(diff) == 3)
```

Comment:
Could move this assert on the size of `diff` to line 48, before directly accessing `diff[0]`, `diff[1]`, `diff[2]` so that if `diff` doesn't return the expected elements we handle that better than panicking from accessing an out of bounds index.